influxdb3_write/src/write_buffer/buffer_segment.rs

```diff
@@ -186,10 +200,18 @@ pub(crate) fn load_buffer_from_segment(
                         Precision::Nanosecond,
                     )?;
 
-                    let db_buffer = buffered_data
-                        .database_buffers
-                        .entry(write.db_name)
-                        .or_default();
+                    let db_name = &write.db_name;
+                    if !buffered_data.database_buffers.contains_key(db_name) {
+                        let db_schema = catalog.db_schema(db_name).expect("db schema should exist");
+                        buffered_data.database_buffers.insert(
+                            db_name.clone(),
+                            DatabaseBuffer {
+                                table_buffers: HashMap::new(),
+                                db_schema,
+                            },
+                        );
+                    }
+                    let db_buffer = buffered_data.database_buffers.get_mut(db_name).unwrap();
```

Comment:
I think you can keep the initial form with using the entry API like so sans borrow check issues:

```suggestion
                    let db_buffer = buffered_data
                        .database_buffers
                        .entry(db_name)
                        .or_insert_with(|| {
                           DatabaseBuffer {
                               table_buffers: HashMap::new(),
                               db_schema: catalog
                                   .db_schema(db_name)
                                   .expect("db schema should exist");
                           }
                        })
```

Reply:
Looking at the PR it seems you had done something similar earlier. Was there a reason you didn't do so here?

Reply:
that doesn't work because `db_name` is a borrowed string, which doesn't work with the entry API. So I did it this way to avoid a string clone, rather than use the entry API.